modules/renter/repair.go

```diff
@@ -193,7 +194,11 @@ func (r *Renter) managedRepairIteration(rs *repairState) {
 		}
 
 		// Ignore workers that have had an upload failure recently.
-		if time.Since(worker.recentUploadFailure) < uploadFailureCooldown {
+		penalty := uint64(worker.consecutiveUploadFailures)
+		if worker.consecutiveUploadFailures > maxConsecutivePenalty {
+			penalty = uint64(maxConsecutivePenalty)
+		}
+		if time.Since(worker.recentUploadFailure) < uploadFailureCooldown*(1<<penalty) {
```

Comment:
oh yeah. If you don't put a reasonable max on the number of doublings, you quickly achieve overflow. That's why I used a doubling-count instead of a max time.